### DISCUSSION TITLE
Blog 4: Klassen

### DISCUSSION THREAD

**Discussion by Max-Rohrhirsch** [DHBW-UMOC/UMOC#8] [Announcements]
### Klassen
In dieser Woche gab es nicht viel Neues. Die Klassenaufgabe wurde erledigt. Ansonsten gab es nichts Neues.
Da wir im Backend nur der Zugriff auf die Datenbank ist und wenig Logik implementiert wird, werden im Backend keine/wenige Klassen verwendet. Später werden eventuell DTOs (Datenklassen) erstellt. Dies stellt sich aber erst später heraus, ob das die Sinnvollste Möglichkeit in Python ist.

Im Frontend hingegen werden einige Klassen verwendet, da alle Komponenten und Service in Angular auch eine Klasse haben/sind. Im Folgenden haben wir die Verknüpfung einiger Klassen im Frontend dargestellt.
Im Laufe der Programmierung werden weitere Klassen entwickelt.
![image](https://github.com/user-attachments/assets/cf2cabaa-2cc0-4fa3-9ab0-38970b46d0a9)

**Comment by OrlandoLegolas** [comment]
Euer Update klingt wie immer sehr interessant. Die Entscheidung, Klassen im Backend sparsam einzusetzen, ist verständlich, wenn die Hauptaufgabe hier nur der Datenbankzugriff ist und wenig Logik benötigt wird. In Python kann man ja gut auch ohne viele Klassen auskommen. Die Idee, später DTOs einzuführen, klingt sinnvoll, falls die Datenstrukturen komplexer werden.

Dass im Frontend mehr Klassen zum Einsatz kommen, passt natürlich gut zu Angular, wo die Struktur über Klassen für Komponenten und Services ohnehin stark vorgegeben ist. Ich bin gespannt, wie sich die Klassenverknüpfungen und das gesamte Design im Frontend entwickeln werden.

Viele Grüße 
Das GymTrackerX Team

**Comment by DonKir15** [comment]
Der Blogpost zeigt, dass euer Team einen pragmatischen Ansatz verfolgt, indem es im Backend auf unnötige Klassen verzichtet und sich auf den Datenbankzugriff konzentriert. Es ist gut, dass ihr flexibel bleibt und die Möglichkeit offenhalt, später DTOs einzuführen, falls die Datenstrukturen komplexer werden. Diese Planung deutet auf ein durchdachtes Vorgehen hin und vermeidet unnötige Komplexität in einer frühen Projektphase.

Im Frontend ist es passend, dass Klassen intensiv genutzt werden, da Angular eine komponentenbasierte Architektur hat, in der jede Komponente und jeder Service als Klasse umgesetzt wird. Die Darstellung der Klassenverknüpfungen im Frontend hilft, die Struktur des Projekts nachvollziehbar darzustellen, und zeigt, dass das Team die Architektur im Blick hat. Es ist ebenfalls positiv, dass ihr plant, das Design weiterzuentwickeln, während das Projekt voranschreitet.

**Comment by bunki1224** [comment]
Dieses Klassendiagramm beeindruckt durch seine klare und gut strukturierte Darstellung. Jede Komponente hat präzise definierte Aufgaben, die Hierarchie und Verbindungen sind logisch und leicht nachvollziehbar. Die Aufteilung zwischen Kontaktliste, Chat-Historie und Textbearbeitung zeigt ein durchdachtes Design, das Effizienz und Benutzerfreundlichkeit fördert. Ein gelungenes Beispiel für strukturiertes Software-Design!
> Euer TripAhead Team (JB)
